clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/client/HttpCloudFoundryClient.java

```diff
@@ -228,7 +240,12 @@ private void refreshTokenIfNecessary() {
   }
 
   private void refreshToken() {
-    token = uaaService.passwordToken("password", user, password, "cf", "");
+    try {
+      token = uaaService.passwordToken("password", user, password, "cf", "");
+    } catch (Exception e) {
+      log.warn("Failed to obtain a token", e);
```

Comment:
Does this double-log the error in the event of failure?

Reply:
Printed back to the user by `hal`:
```
! ERROR Failed to fetch spaces while validating account
  'account_name': Cloud Foundry API returned with error(s): Unable to retrieve
  spaces.
```

Printed in the logs for the same error:
```
2019-06-03 13:36:11.179  WARN 20209 --- [       Thread-6] c.n.s.c.c.client.HttpCloudFoundryClient  : Failed to obtain a token

retrofit.RetrofitError: 404 Not Found
        at retrofit.RetrofitError.httpError(RetrofitError.java:40) ~[retrofit-1.9.0.jar:na]
        at retrofit.RestAdapter$RestHandler.invokeRequest(RestAdapter.java:388) [retrofit-1.9.0.jar:na]
        at retrofit.RestAdapter$RestHandler.invoke(RestAdapter.java:240) [retrofit-1.9.0.jar:na]
        at com.sun.proxy.$Proxy135.passwordToken(Unknown Source) ~[na:na]
        at com.netflix.spinnaker.clouddriver.cloudfoundry.client.HttpCloudFoundryClient.refreshToken(HttpCloudFoundryClient.java:244) [clouddriver-cloudfoundry.jar:na]
        at com.netflix.spinnaker.clouddriver.cloudfoundry.client.HttpCloudFoundryClient.refreshTokenIfNecessary(HttpCloudFoundryClient.java:238) [clouddriver-cloudfoundry.jar:na]
        at com.netflix.spinnaker.clouddriver.cloudfoundry.client.HttpCloudFoundryClient.access$000(HttpCloudFoundryClient.java:53) [clouddriver-cloudfoundry.jar:na]
        at com.netflix.spinnaker.clouddriver.cloudfoundry.client.HttpCloudFoundryClient$1.intercept(HttpCloudFoundryClient.java:77) [clouddriver-cloudfoundry.jar:na]
        at retrofit.RestAdapter$RestHandler.invokeRequest(RestAdapter.java:300) [retrofit-1.9.0.jar:na]
        at retrofit.RestAdapter$RestHandler.invoke(RestAdapter.java:240) [retrofit-1.9.0.jar:na]
        at com.sun.proxy.$Proxy137.all(Unknown Source) [na:na]
        at com.netflix.spinnaker.clouddriver.cloudfoundry.client.Spaces.lambda$all$0(Spaces.java:69) [clouddriver-cloudfoundry.jar:na]
        at com.netflix.spinnaker.clouddriver.cloudfoundry.client.CloudFoundryClientUtils.lambda$collectPageResources$4(CloudFoundryClientUtils.java:86) ~[clouddriver-cloudfoundry.jar:na]
        at com.netflix.spinnaker.clouddriver.cloudfoundry.client.CloudFoundryClientUtils.safelyCall(CloudFoundryClientUtils.java:43) ~[clouddriver-cloudfoundry.jar:na]
        at com.netflix.spinnaker.clouddriver.cloudfoundry.client.CloudFoundryClientUtils.collectPageResources(CloudFoundryClientUtils.java:86) ~[clouddriver-cloudfoundry.jar:na]
        at com.netflix.spinnaker.clouddriver.cloudfoundry.client.Spaces.all(Spaces.java:69) [clouddriver-cloudfoundry.jar:na]
        at com.netflix.spinnaker.halyard.config.validate.v1.providers.cloudfoundry.CloudFoundryAccountValidator.validate(CloudFoundryAccountValidator.java:83) ~[halyard-config.jar:na]
        at com.netflix.spinnaker.halyard.config.validate.v1.providers.cloudfoundry.CloudFoundryProviderValidator.lambda$validate$0(CloudFoundryProviderValidator.java:29) ~[halyard-config.jar:na]
        at java.util.ArrayList.forEach(ArrayList.java:1257) ~[na:1.8.0_212]
        at com.netflix.spinnaker.halyard.config.validate.v1.providers.cloudfoundry.CloudFoundryProviderValidator.validate(CloudFoundryProviderValidator.java:29) ~[halyard-config.jar:na]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_212]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_212]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_212]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_212]
        at com.netflix.spinnaker.halyard.config.validate.v1.ValidatorCollection.runMatchingValidators(ValidatorCollection.java:79) ~[halyard-config.jar:na]
        at com.netflix.spinnaker.halyard.config.validate.v1.ValidatorCollection.runAllValidators(ValidatorCollection.java:53) ~[halyard-config.jar:na]
        at com.netflix.spinnaker.halyard.config.services.v1.ValidateService.recursiveValidate(ValidateService.java:55) ~[halyard-config.jar:na]
        at com.netflix.spinnaker.halyard.config.services.v1.ValidateService.recursiveValidate(ValidateService.java:63) ~[halyard-config.jar:na]
        at com.netflix.spinnaker.halyard.config.services.v1.ValidateService.recursiveValidate(ValidateService.java:63) ~[halyard-config.jar:na]
        at com.netflix.spinnaker.halyard.config.services.v1.ValidateService.recursiveValidate(ValidateService.java:63) ~[halyard-config.jar:na]
        at com.netflix.spinnaker.halyard.config.services.v1.ValidateService.validateMatchingFilter(ValidateService.java:49) ~[halyard-config.jar:na]
        at com.netflix.spinnaker.halyard.config.services.v1.AccountService.validateAccount(AccountService.java:123) ~[halyard-config.jar:na]
        at com.netflix.spinnaker.halyard.controllers.v1.AccountController.lambda$addAccount$12(AccountController.java:155) ~[main/:na]
        at com.netflix.spinnaker.halyard.core.DaemonResponse$UpdateRequestBuilder.build(DaemonResponse.java:156) ~[halyard-core.jar:na]
        at com.netflix.spinnaker.halyard.core.tasks.v1.TaskRepository.lambda$submitTask$1(TaskRepository.java:48) ~[halyard-core.jar:na]
        at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_212]
```

Isn't logged twice per se unless I misunderstood the question.

Reply:
OK, great. Just wanted to check.